src/util/animation/ColorAnimation.ts

```diff
@@ -0,0 +1,65 @@
+import { Color } from '../../color';
+import { TColorAlphaSource } from '../../color/color.class';
+import { noop } from '../../constants';
+import { AnimationBase } from './AnimationBase';
+import {
+  ColorAnimationOptions,
+  TColorEasingRateFunction,
+  TOnAnimationChangeCallback,
+} from './types';
+
+const wrapColorCallback =
+  <R>(callback: TOnAnimationChangeCallback<string, R>) =>
+  (rgba: TColorAlphaSource, valueRate: number, durationRate: number) =>
+    callback(new Color(rgba).toRgba(), valueRate, durationRate);
+
+const defaultColorEasingRate: TColorEasingRateFunction = (
+  currentTime,
+  duration
+) => 1 - Math.cos((currentTime / duration) * (Math.PI / 2));
+
+export class ColorAnimation extends AnimationBase<TColorAlphaSource> {
+  readonly colorEasing: TColorEasingRateFunction;
+  constructor({
+    startValue,
+    endValue,
+    easing = (currentTime, startValue, byValue) => startValue + byValue,
+    colorEasing = defaultColorEasingRate,
+    onChange = noop,
+    onComplete = noop,
+    abort = noop,
+    ...options
+  }: ColorAnimationOptions) {
+    const startColor = new Color(startValue).getSource();
+    const endColor = new Color(endValue).getSource();
+    super({
+      ...options,
+      startValue: startColor,
+      endValue: endColor,
+      byValue: endColor.map(
+        (value, i) => value - startColor[i]
+      ) as TColorAlphaSource,
+      easing,
+      onChange: wrapColorCallback(onChange),
+      onComplete: wrapColorCallback(onComplete),
+      abort: wrapColorCallback(abort),
+    });
+    this.colorEasing = colorEasing;
+  }
+  protected calculate(currentTime: number) {
+    const changeRate = this.colorEasing(currentTime, this.duration);
```

Comment:
I think it makes sense moving the color easing logic to the default easing function
Then ColorAnimation is identical to the rest

Reply:
fd763b7a8